src/cpu/dyn_cache.h

```diff
@@ -738,21 +788,32 @@ static void cache_init(bool enable) {
 				cache_blocks[i].cache.next = &cache_blocks[i + 1];
 			}
 		}
-		if (cache_code_start_ptr==NULL) {
+		if (cache_code_start_ptr==nullptr) {
 			// allocate the code cache memory
 #if defined (WIN32)
-			cache_code_start_ptr=(Bit8u*)VirtualAlloc(0,CACHE_TOTAL+CACHE_MAXSIZE+PAGESIZE_TEMP-1+PAGESIZE_TEMP,
-				MEM_COMMIT,PAGE_EXECUTE_READWRITE);
+			cache_code_start_ptr=static_cast<uint8_t *>(VirtualAlloc(0,cache_code_size,
+				MEM_COMMIT,PAGE_EXECUTE_READWRITE));
 			if (!cache_code_start_ptr)
-				cache_code_start_ptr=(Bit8u*)malloc(CACHE_TOTAL+CACHE_MAXSIZE+PAGESIZE_TEMP-1+PAGESIZE_TEMP);
+				cache_code_start_ptr=static_cast<uint8_t *>(malloc(cache_code_size));
 #else
-			cache_code_start_ptr=(Bit8u*)malloc(CACHE_TOTAL+CACHE_MAXSIZE+PAGESIZE_TEMP-1+PAGESIZE_TEMP);
+			int map_flags = MAP_PRIVATE | MAP_ANON;
+			int prot_flags = PROT_READ | PROT_WRITE | PROT_EXEC;
+#if defined(HAVE_MAPJIT)
+			map_flags |= MAP_JIT;
 #endif
+#if defined(HAVE_MMAP)
+			cache_code_start_ptr=static_cast<uint8_t *>(mmap(nullptr, cache_code_size, prot_flags, map_flags, -1, 0));
+			if (cache_code_start_ptr == MAP_FAILED)
+				E_Exit("Allocating dynamic core cache memory failed with errno %d", errno);
+#else
+			cache_code_start_ptr=static_cast<uint8_t *>(malloc(cache_code_size));
 			if (!cache_code_start_ptr)
 				E_Exit("Allocating dynamic core cache memory failed");
-
+#endif
+#endif
 			// align the cache at a page boundary
-			cache_code = (Bit8u *)(((Bitu)cache_code_start_ptr +
+			cache_code = reinterpret_cast<uint8_t *>(
+									(reinterpret_cast<uintptr_t>(cache_code_start_ptr) +
```

Comment:
To indent this `(reinterpret_cast<uintptr_t> ...` line, use tabs until it lines up with the `cache_code` above, then use spaces to indent it inward to the desired alignment..

This approach of "tabs-for the outer whitespace", then "spaces-for-inner alignment", ensures that any tab-width will work properly, and inner-alignment will also be correct.